bijection-util/src/test/scala/com/twitter/bijection/twitter_util/UtilBijectionLaws.scala

```diff
@@ -16,72 +16,92 @@ limitations under the License.
 
 package com.twitter.bijection.twitter_util
 
-import com.twitter.bijection.{ CheckProperties, BaseProperties, Bijection }
+import com.google.common.util.concurrent.Futures
+import com.twitter.bijection.{ CheckProperties, BaseProperties }
 import com.twitter.io.Buf
 import com.twitter.util.{ Future => TwitterFuture, Try => TwitterTry, Await => TwitterAwait }
 import java.lang.{ Integer => JInt, Long => JLong }
+import java.util.concurrent.{ Future => JavaFuture }
 import org.scalacheck.Arbitrary
-import org.scalatest.{ PropSpec, MustMatchers }
-import org.scalatest.prop.PropertyChecks
+import org.scalatest.BeforeAndAfterAll
 
-import org.scalacheck.Prop.forAll
 import scala.concurrent.{ Future => ScalaFuture, Await => ScalaAwait }
 import scala.concurrent.duration.Duration
 import scala.util.{ Try => ScalaTry }
 import scala.concurrent.future
 import scala.concurrent.ExecutionContext.Implicits.global
 
-class UtilBijectionLaws extends CheckProperties with BaseProperties {
+class UtilBijectionLaws extends CheckProperties with BaseProperties with BeforeAndAfterAll {
   import UtilBijections._
 
-  protected def toOption[T](f: TwitterFuture[T]): Option[T] = TwitterTry(TwitterAwait.result(f)).toOption
+  protected def toOption[T](f: TwitterFuture[T]): Option[T] =
+    TwitterTry(TwitterAwait.result(f)).toOption
 
-  protected def toOption[T](f: ScalaFuture[T]): Option[T] = TwitterTry(ScalaAwait.result(f, Duration.Inf)).toOption
+  protected def toOption[T](f: ScalaFuture[T]): Option[T] =
+    TwitterTry(ScalaAwait.result(f, Duration.Inf)).toOption
 
-  implicit def futureArb[T: Arbitrary] = arbitraryViaFn[T, TwitterFuture[T]] { TwitterFuture.value(_) }
+  protected def toOption[T](f: JavaFuture[T]): Option[T] =
+    TwitterTry(f.get()).toOption
+
+  implicit def futureArb[T: Arbitrary] = arbitraryViaFn[T, TwitterFuture[T]] { TwitterFuture.value }
   implicit def scalaFutureArb[T: Arbitrary] = arbitraryViaFn[T, ScalaFuture[T]] { future(_) }
+  implicit def javaFutureArb[T: Arbitrary] =
+    arbitraryViaFn[T, JavaFuture[T]] { Futures.immediateFuture }
```

Comment:
I'm not sure this is worth entail the dependency.  How do you feel about rolling your own with `FutureTask`?